includes/multi-currency/class-multi-currency.php

```diff
@@ -437,6 +436,20 @@ public function get_selected_currency(): Currency {
 		return $this->get_enabled_currencies()[ $code ] ?? $this->default_currency;
 	}
 
+	/**
+	 * This action is triggered when the WooCommerce store settings are updated. It will check if the store currency has changed,
+	 * and if it has, clear the cache and reinitialize the currencies, which will update the default currency, available currencies and
+	 * enabled currencies.
+	 *
+	 * @return void
+	 */
+	public function update_default_currency() {
```

Comment:
I should have checked your PR before working on the notice. I went with a more aggressive approach to checking if the currency changed: https://github.com/Automattic/woocommerce-payments/pull/2322/files#diff-b17bf67925ece08a12ad748cd870705e5d668d3fa19214040f4841daaf712cdcR647

My thought was that if something outside of the General settings page changed the currency, we wouldn't know, and that's why I didn't queue this off of the `woocommerce_update_options_general` action. Do you think it would be better to check based off of the action, and if we get reports of issues address those as they come?

Reply:
I like your approach for doing this! One of my concerns while working on this PR was that only checking when the option gets changed on the settings page may not cover all cases where the store currency gets changed.

I think we can probably reconcile our two approaches, so that the logic I have here to query the server and fetch new rates uses the check you implemented in your PR and updates if needed.

I'll take a look at updating my PR after the NIRL meetup! :)